dm/syncer/checkpoint.go

```diff
@@ -561,12 +677,14 @@ func (cp *RemoteCheckPoint) FlushPointsExcept(tctx *tcontext.Context, exceptTabl
 		return err
 	}
 
-	cp.globalPoint.flush()
+	cp.Lock()
```

Comment:
seems `binlogPoint` has its own lock so doesn't need the checkpoint lock.

And `needFlushSafeModeExitPoint` will be written by `checkpointFlushWorker.Run` and `SaveSafeModeExitPoint`, `SaveSafeModeExitPoint` is called in `Syncer.Run` so should be protected by checkpoint lock